Add Hosted VLLM rerank provider integration

PhilipAD opened the pull request:
## Add Hosted VLLM rerank provider integration

This commit implements the Hosted VLLM rerank provider integration for LiteLLM. 

The integration includes: 

- Adding Hosted VLLM as a supported rerank provider in the main rerank function 
- Implementing the HostedVLLMRerank handler class for making API requests 
- Creating a transformation class to convert Hosted VLLM responses to LiteLLM's standardized format 
- The integration supports both synchronous and asynchronous rerank operations. 
- API credentials can be provided directly or through environment variables (HOSTED_VLLM_API_KEY and HOSTED_VLLM_API_BASE). 

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the `tests/litellm/` directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [ ] My PR passes all unit tests on (`make test-unit`)[https://docs.litellm.ai/docs/extras/contributing_code]
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type
🆕 New Feature

## Changes
This PR implements Hosted VLLM rerank integration in LiteLLM. Changes include:

1. Added Hosted VLLM as a supported provider in the main rerank function
2. Implemented HostedVLLMRerank handler for API requests with both sync and async support
3. Created transformation logic to standardize Hosted VLLM responses
4. Added proper environment variable support (HOSTED_VLLM_API_KEY and HOSTED_VLLM_API_BASE)
5. Implemented error handling for missing credentials and invalid responses

The integration follows the same patterns as existing rerank providers while addressing the specific requirements of the Hosted VLLM API.

Commit:
Add Hosted VLLM rerank provider integration

This commit implements the Hosted VLLM rerank provider integration for LiteLLM. The integration includes:
Adding Hosted VLLM as a supported rerank provider in the main rerank function
Implementing the HostedVLLMRerank handler class for making API requests
Creating a transformation class to convert Hosted VLLM responses to LiteLLM's standardized format
The integration supports both synchronous and asynchronous rerank operations. API credentials can be provided directly or through environment variables (HOSTED_VLLM_API_KEY and HOSTED_VLLM_API_BASE).
Notable features:
Proper error handling for missing credentials
Standard response transformation
Support for common rerank parameters (top_n, return_documents, etc.)
Proper token usage tracking
This expands LiteLLM's rerank provider ecosystem to include Hosted VLLM alongside existing providers like Cohere, Together AI, Azure AI, and Bedrock.

krrishdholakia commented on litellm/llms/hosted_vllm/rerank/handler.py:
```diff
@@ -0,0 +1,103 @@
+from litellm.types.rerank import RerankRequest, RerankResponse
+
+
+class HostedVLLMRerank(BaseLLM):
```
there is no need for this handler. please follow the cohere integration, and use the base_llm_http_handler - https://github.com/BerriAI/litellm/blob/3875df666b8a11819eda86fdc35d582be4bd8db6/litellm/rerank_api/main.py#L183

krrishdholakia commented on litellm/rerank_api/main.py:
```diff
@@ -321,6 +323,37 @@ def rerank(  # noqa: PLR0915
+                    "api_base must be provided for Hosted VLLM rerank. Set in call or via HOSTED_VLLM_API_BASE env var."
+                )
+
+            response = hosted_vllm_rerank.rerank(
```
replace with - https://github.com/BerriAI/litellm/blob/3875df666b8a11819eda86fdc35d582be4bd8db6/litellm/rerank_api/main.py#L183

krrishdholakia commented on litellm/llms/hosted_vllm/rerank/transformation.py:
```diff
@@ -0,0 +1,65 @@
+)
+
+
+class HostedVLLMRerankConfig:
```
this class is invalid, as it does not inherit from the basererankconfig - https://github.com/BerriAI/litellm/blob/3875df666b8a11819eda86fdc35d582be4bd8db6/litellm/llms/cohere/rerank/transformation.py#L16

Since vllm is cohere compatible, the implementation should just inherit from that class, like azure_ai does - https://github.com/BerriAI/litellm/blob/3875df666b8a11819eda86fdc35d582be4bd8db6/litellm/llms/azure_ai/rerank/transformation.py

krrishdholakia reviewed (changes requested):
Refactor to follow similar flow as azure ai implementation, which is also cohere compatible - https://github.com/BerriAI/litellm/blob/3875df666b8a11819eda86fdc35d582be4bd8db6/litellm/llms/azure_ai/rerank/transformation.py

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=9249) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=9249) before we can accept your contribution.<br/><hr/>**Philip D'Souza** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=9249) it.</sub>

todd0806 commented:
I need the commit too...Please merge it , thanks.

krrishdholakia commented:
Hey @todd0806 waiting on @PhilipAD to implement the changes. 

In the meanwhile, you can try using the vllm passthrough route - https://docs.litellm.ai/docs/pass_through/vllm, if you're on the proxy.

Commit:
refactor(rerank): use base_llm_http_handler for hosted_vllm rerank

- Replace custom HostedVLLMRerank handler with base_llm_http_handler
- Implement proper HostedVLLMRerankConfig inheriting from BaseRerankConfig
- Follow Cohere-compatible implementation pattern
- Clean up unnecessary comments

Commit:
Fix lint errors in hosted_vllm rerank transformer: remove unused imports

Commit:
Fix linting errors in rerank transformation modules

Commit:
fix: resolve type errors in Hosted VLLM rerank module

PhilipAD commented:
Changes implemented! PR is ready for review. I've refactored the hosted_vllm rerank to use the base_llm_http_handler following the Cohere-compatible pattern. @krrishdholakia

PhilipAD committed:
Merge branch 'BerriAI:main' into vllm_rerank

ishaan-jaff reviewed (changes requested):
please add a mapped test in tests_litellm/llms/hosted_vllm 

Ping me here once done, will oversee it

krrishdholakia commented:
Closing as this was added via https://github.com/BerriAI/litellm/pull/12738
